nixos/modules/programs/steam.nix

```diff
@@ -20,6 +22,11 @@ let
       Exec=${steam-gamescope}/bin/steam-gamescope
       Type=Application
     '').overrideAttrs (_: { passthru.providedSessions = [ "steam" ]; });
+
+  protontricks = cfg.protontricks.package.overrideAttrs (prev: {
+    makeWrapperArgs = prev.makeWrapperArgs ++
+      (lib.optional (cfg.extraCompatPackages != [ ]) "--set STEAM_EXTRA_COMPAT_TOOLS_PATHS ${extraCompatPaths}");
+  });
```

Comment:
It'd be better if protontricks provided a proper override option for this. `overrideAttrs` isn't really a stable interface.

Reply:
I'm not sure what you mean by that / how it should be implemented. Do you mean the derivation should contain an argument for extraSteamCompatPaths, and we should use override here instead?

Reply:
Yup that's precisely it. You'd basically define an API that's quite a bit more stable than overrideAttrs.